wandb/history.py

```diff
@@ -142,20 +141,8 @@ def _index(self, row):
         self._steps += 1
 
     def _transform(self):
-        """Transforms media classes into the proper format before writing"""
-        for key, val in six.iteritems(self.row):
-            if isinstance(val, media.Image):
-                val = [val]
-            if isinstance(val, collections.Sequence) and len(val) > 0:
-                is_image = [isinstance(v, media.Image) for v in val]
-                if all(is_image):
-                    self.row[key] = media.Image.transform(val, self.out_dir,
-                                                          "{}_{}.jpg".format(key, self.row["_step"]))
-                elif any(is_image):
-                    raise ValueError(
-                        "Mixed media types in the same list aren't supported")
-            elif isinstance(val, data_types.Histogram):
-                self.row[key] = data_types.Histogram.transform(val)
+        """Transforms special classes into the proper format before writing"""
+        self.row = data_types.Base.convert(self.row)
```

Comment:
should `data_types.Base.convert` really be a method of the base class? Since it's never called by an instance of a subclass, it seems like it's more of a factory function that should live separately from each individual class -- something like `data_types.create_data_object_for_row(self.row)`